### DISCUSSION TITLE
Other IR protocols than listad for HVAC devices

### DISCUSSION THREAD

**Discussion by cociweb** [arendst/Tasmota#12686] [Q&A] [answered]
Hello,

I would like to build my custom ir build with the support of IRremoteESP8266. I've searched a lot for tutorials but this version of tasmota is very 'underdocumented'. 

I would like to specify only one protocol from [this](https://tasmota.github.io/docs/Tasmota-IR/#sending-irhvac-commands) list (only TCL112AC) to allocate the minimum flash and memory for sending and receiving IRHVAC commands. is there any available tutorial for this specific case? If not, can somebody guide through me what is essential for the custom ir build to filter out only a specific send/receive protocol (over mqtt)?

Thank's in advance.

**Comment by Jason2866** [comment, marked as answer]
Yes, we need help for adding this in the docu. The docu is Open Source too.
You find a short description here https://github.com/arendst/Tasmota/blob/development/tasmota/my_user_config.h#L795-L823
This defines are used here https://github.com/arendst/Tasmota/blob/development/lib/lib_basic/IRremoteESP8266/IRremoteESP8266/src/IRremoteESP8266.h#L63-L770
